community/bolt/src/main/java/org/neo4j/bolt/security/auth/BasicAuthentication.java

```diff
@@ -62,17 +62,17 @@ public AuthenticationResult authenticate( Map<String,Object> authToken ) throws
         String password = safeCast( CREDENTIALS, authToken );
         if ( authToken.containsKey( NEW_CREDENTIALS ) )
         {
-            return update( user, password, safeCast( NEW_CREDENTIALS, authToken ) );
+            return update( user, password, safeCast( NEW_CREDENTIALS, authToken ), authToken );
         }
         else
         {
-            return authenticate( user, password );
+            return authenticate( user, password, authToken );
         }
     }
 
-    private AuthenticationResult authenticate( String user, String password ) throws AuthenticationException
+    private AuthenticationResult authenticate( String user, String password, Map<String,Object> authToken ) throws AuthenticationException
```

Comment:
Yes. The explanation for the current form is that it was part of a last-minute backup plan (see below), and as such the idea was to minimize last minute code changes. But since that plan was abandoned we will reopen this with a proper API when needed.